src/main/kotlin/io/stud/forest/book_a_cinema_ticket_app/application/dto/RepertoireDto.kt

```diff
@@ -0,0 +1,43 @@
+package io.stud.forest.book_a_cinema_ticket_app.application.dto
+
+
+import com.fasterxml.jackson.annotation.JsonPropertyOrder
+import io.stud.forest.book_a_cinema_ticket_app.domain.entities.Screening
+import io.stud.forest.book_a_cinema_ticket_app.domain.repositories.Id
+import java.time.LocalDateTime
+
+@JsonPropertyOrder("movieTitle", "timetable")
+data class RepertoireDto(
+        val movieTitle: String,
+        val timetable: List<TimeTableItem>,
+) {
+    companion object {
+        fun listOf(screenings: List<Screening>): List<RepertoireDto> =
+                screenings
+                        .groupBy { it.movie.title }
+                        .map(sortEachGroupByStartTime)
+                        .map(mapGroupToDto)
+                        .sortedBy { it.movieTitle }
+
+        private val mapGroupToDto: (Map.Entry<String, List<Screening>>) -> RepertoireDto = { titleToScreenings ->
+            RepertoireDto(
+                    titleToScreenings.key,
+                    titleToScreenings.value.map {
+                        TimeTableItem(
+                                it.id,
+                                concatScreeningDateAndTime(it))
+                    })
+        }
+
+        private val concatScreeningDateAndTime: (Screening) -> LocalDateTime = { it -> LocalDateTime.of(it.day, it.startTime) }
+        private val sortEachGroupByStartTime: (Map.Entry<String, List<Screening>>) -> Map.Entry<String, List<Screening>> = { group ->
+            group.value.sortedBy { it.startTime }.let { group }
```

Comment:
Hmmm, nie jestem pewien, ale, czy sortedBy nie zwraca nowej listy zamiast modyfikować istniejącą? Natomiast na koniec zwracasz to co na wejściu, więc czy ta funkcja coś w ogóle zrobi? Czy nie powinniśmy uzyć funkcji sortBy?

Reply:
> czy sortedBy nie zwraca nowej listy zamiast modyfikować istniejącą?
Dokładnie tak.
![sortBy](https://user-images.githubusercontent.com/24223222/128170096-a81bf10e-9ae5-4ee5-af28-6c670f68c314.PNG)

> na koniec zwracasz to co na wejściu, więc czy ta funkcja coś w ogóle zrobi?
Zgadza się, zwracany jest parametr wejściowy.

> Czy nie powinniśmy uzyć funkcji sortBy?
Niewątpliwie trzeba poprawić implementację, ale lista która jest do posortowania jest niemutowalna, więc nie udostępnia sortBy, na szybko nie znalazłem też możliwości stworzenia nowego Map.Enrty, więc może się okazać, że filmy trzeba będzie posortować jeszcze przed rozbiciem na grupy.

Reply:
Na liście można też chyba zawołać coś w stylu `toMutableList`, jeśli potrzebujesz. Ale ogólnie dałoby się to zapisać lepiej, np  (pseudokod, piszę tylko na githubie):
```
group.key to group.value.sortedBy { it.startTime }
```
P. S. Przydałby się pewnie na to test i wszystko byłoby jasne :)